src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -84,11 +90,73 @@ public static void ToCharsBuffer(byte value, Span<char> buffer, int startingInde
             buffer[startingIndex] = (char)(packedResult >> 8);
         }
 
+#if SYSTEM_PRIVATE_CORELIB
+    private static void EncodeToUtf16_Ssse3(ReadOnlySpan<byte> bytes, Span<char> chars, Casing casing)
+    {
+        int pos = 0;
```

Comment:
```suggestion
        nint pos = 0;
```
to avoid some `movsxd` in the `Unsafe`-methods (to pick the IntPtr-overload) (except current JIT can handle this now)?

Reply:
For some reason codegen is slightly worse with nint.

Reply:
https://www.diffchecker.com/Dcujlda7 (nint is on the right)